app/views/samples/_upload_results_js.haml

```diff
@@ -0,0 +1,137 @@
+        else 
+            document.getElementById(`bulk_process_csv`).disabled = true;
+    }
+
+    function handleFileSelect(evt) {
+        // We use the 1st file from the FileList list
+        let f = evt.target.files[0];
+        let reader = new FileReader();
+
+        // Closure to capture the file information.
+        reader.onload = (function(f) {
+            return async(e) => {
+                // Parse the input file
+                var text =  e.target.result
+                var lines = text.split(/\r\n|\r|\n/); // tolerate both Windows and Unix linebreaks
+                var samples_param = [] // params for GET call
+                var samples = [] // samples uuids
+
+                lines.forEach(function(line) { 
+                    if (line != ''){
+                        samples_param.push( 'uuids[]=' + line.split(',')[0] );
+                        samples.push( line.split(',')[0] );
+                    } 
+                });
+                
+                // Retrieve found uuids
+                var context = getUrlParameter('context');
+                var url = '/samples/existing_uuids?context='+context+"&"+samples_param.join("&")
+                var found = await fetch(url)
+                    .then((response) => response.json())
+                    .then((r) => {
+                        return r.message;
+                })
+                // Compute not found ones
+                var not_found = samples.filter(function(x) {
+                    return found.indexOf(x) < 0;
+                });
+                
+                // Create row with filename and upload info
+                var upload_info = $('<div>', { class: `batches-samples` })
+                    .append( $('<div>', { class: `samples-row` }) 
+                    .append( $('<div>', { class: `samples-item` })
+                        .append( $('<div>', { 
+                            class: `icon-circle-minus icon-gray `,
+                            id: `remove_file_${evt.target.nowParam}`
+                        }) )
+                        .append( "&nbsp;" + f.name ) 
+                    )
+                    .append( (found.length == samples.length) ? 
+                        $('<div>', { class: `samples-row-action` })
+                        .append( `(${samples.length} Samples) ` )
+                        .append( $('<div>', { class: `icon-checkmark  bigger` }) )  :
+                        $('<div>', { class: `samples-row-action ttip input-required`,
+                                     id: `not_found_mess_${evt.target.nowParam}`, 
+                                     title: `${not_found.join('\n')}` 
+                                    })
+                            .append( `${samples.length} Samples, (${(samples.length-found.length)} sample UUID(s) not found) ` ) 
+                            .append( $('<div>', { class: `icon-alert icon-red bigger` }) )
+                        )
+                    );
```

Comment:
This could be abstracted with a `<template>` element. You'd only have to set a few values in the cloned DOM. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template

```html
<template id="csvFileRow">
  <div class="">
    <span class="samples-count"></span> Samples
  </div>
</template>
```

```js
var template = document.getElementById("csvFileRow").content;
var fragment = template.cloneNode(true);
fragment.querySelector(".samples-count").textContent = samples.length;
```

Reply:
Note that you don't have to use jQuery. The DOM is nice/standard enough nowadays.